src/backend/ens.nim

```diff
@@ -1,216 +1,21 @@
-import sequtils
-import strutils
-import nimcrypto
 import json
-import json_serialization
-import tables
-import stew/byteutils
-import web3/[ethtypes, conversions], stint
-import chronicles, libp2p/[multihash, multicodec, cid]
+import ./core, ./response_type
 
+export response_type
 
-import ./statusgo_backend/wallet
-import ./statusgo_backend/accounts as status_accounts
-import ./statusgo_backend/settings as status_settings
-import ./statusgo_backend_new/ens as status_ens
+proc resolver*(username: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+  # TODO: Use a real chain id
+  let payload = %* [1, username]
 
-import ./types/[transaction, setting, rpc_response, network_type, network, profile]
-import ./utils
-import ./transactions
-import ./eth/contracts
+  return core.callPrivateRPC("ens_resolver", payload)
 
-const domain* = ".stateofus.eth"
+proc contentHash*(username: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+  # TODO: Use a real chain id
+  let payload = %* [1, username]
 
-proc userName*(ensName: string, removeSuffix: bool = false): string =
-  if ensName != "" and ensName.endsWith(domain):
-    if removeSuffix:
-      result = ensName.split(".")[0]
-    else:
-      result = ensName
-  else:
-    if ensName.endsWith(".eth") and removeSuffix:
-      return ensName.split(".")[0]
-    result = ensName
+  return core.callPrivateRPC("ens_contentHash", payload)
 
-proc addDomain*(username: string): string =
-  if username.endsWith(".eth"):
-    return username
-  else:
-    return username & domain
-
-proc hasNickname*(contact: Profile): bool = contact.localNickname != ""
-
-proc userNameOrAlias*(contact: Profile, removeSuffix: bool = false): string =
-  if(contact.ensName != "" and contact.ensVerified):
-    result = "@" & userName(contact.ensName, removeSuffix)
-  elif(contact.localNickname != ""):
-    result = contact.localNickname
-  else:
-    result = contact.alias
-
-proc resolver*(username: string): string =
```

Comment:
If you remove those functions, it means they are not used but that means that the call are still made directly via eth_call and not using the ens backend right? 

I cannot see migration of those function

Reply:
I may have put my comment on the wrong line, but for example owner proc?

Reply:
You're right, comparing the master code and master's code and this one, I am missing a lot of stuff. I'm adding them back, but it's not exactly as copy paste, so it will take some time. I'll re-ping when I'm done